src/lib/components/ToastAlert/useToastManager.ts

```diff
@@ -0,0 +1,111 @@
+import { useCallback, useContext, useMemo } from "react"
+import { AlertContext } from "./ToastAlertProvider"
+import { TAlert, TNewToastOptions } from "./ToastAlert.model"
+import { originCalculation } from "./ToastAlert.utils"
+
+export function useToastManager() {
+  const { dispatch, toastAlerts, origin } = useContext(AlertContext)
+
+  const defaultDuration = useMemo(() => 7000, [])
+  const defaultIntent = useMemo(() => "primary" as const, [])
+  const originTranslate = useMemo(() => originCalculation(origin), [origin])
+
+  const removeToastTimeout = useCallback(
+    (id: TAlert["id"] | undefined) => {
+      const alert = toastAlerts.find(alert => alert.id === id)
```

Comment:
Remove `| undefined` for the id type.  There is no code path that can result in undefined to this function.